src/Microsoft.ML.Data/MLContext.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using System;
+
+namespace Microsoft.ML
+{
+    /// <summary>
+    /// The <see cref="MLContext"/> is a starting point for all ML.NET operations. It is instantiated by user,
+    /// provides mechanisms for logging and entry points for training, prediction, model operations etc.
+    /// </summary>
+    public sealed class MLContext : IHostEnvironment
+    {
+        // REVIEW: consider making LocalEnvironment and MLContext the same class instead of encapsulation.
```

Comment:
I still don't fully understand our plan here with `LocalEnvironment` and `ConsoleEnvironment`. How could I ever use a `ConsoleEnvironment` with this new `MLContext` class?

And if the answer is "you can't - you'd use ConsoleEnvironment directly", I think that is a pretty poor answer because then my whole API paradigm changes. I can no longer use `BinaryClassificationContext`, `Transforms`, etc. from the `MLContext`.

Reply:
What do you think about the answer: "you can't use ConsoleEnvironment because it is internal and only used by MAML"?

---
In reply to: [225986842](https://github.com/dotnet/machinelearning/pull/1252#discussion_r225986842) [](ancestors = 225986842)

Reply:
That would be fine with me. For now, I would think it's completely fine to have our custom console printing internal to the commandline tool

---
In reply to: [225989180](https://github.com/dotnet/machinelearning/pull/1252#discussion_r225989180) [](ancestors = 225989180,225986842)

Reply:
https://github.com/dotnet/machinelearning/issues/1284

Reply:
Thanks.

---
In reply to: [226052093](https://github.com/dotnet/machinelearning/pull/1252#discussion_r226052093) [](ancestors = 226052093)